external_search.py

```diff
@@ -2203,32 +2217,58 @@ def create_search_doc(self, query_string, filter=None, pagination=None, debug=Fa
 
     def query_works(self, query_string, filter, pagination, debug=False, return_raw_results=False, search=None):
         self.queries.append((query_string, filter, pagination, debug, return_raw_results))
+        docs = self.docs.values()
+        if pagination:
+            start_at = 0
+            if isinstance(pagination, SortKeyPagination):
```

Comment:
This bit is pretty ugly but it lets the mocked `query_works` work with both regular `Pagination` and `SortKeyPagination`.